MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionCommitManager.cs

```diff
@@ -33,6 +36,34 @@ public bool ShouldCommitCompletion (IAsyncCompletionSession session, SnapshotPoi
 
 		public CommitResult TryCommit (IAsyncCompletionSession session, ITextBuffer buffer, CompletionItem item, char typedChar, CancellationToken token)
 		{
+			string insertionText;
+			var completionKind = item.Properties.PropertyList[item.Properties.PropertyList.Count - 1].Value;
+			switch (completionKind) {
+			case MSBuildSpecialCommitKind.Element:
+				if (item.InsertText.ToLower () == "packagereference") {
+					insertionText = item.InsertText + '/' + '>';
+					Insert (session, buffer, insertionText);
+					ShiftCaret (session, 2, CaretDirection.Left);
+				} else {
+					insertionText = item.InsertText + '>' + '<' + '/' + item.InsertText + '>';
+					Insert (session, buffer, insertionText);
+					ShiftCaret (session, item.InsertText.Length + 3, CaretDirection.Left);
+				}
+				return CommitResult.Handled;
+
+			case MSBuildSpecialCommitKind.Attribute:
+				insertionText = item.InsertText + '=' + '"' + '"';
+				Insert (session, buffer, insertionText);
+				ShiftCaret (session, 1, CaretDirection.Left);
+				return CommitResult.Handled;
+
+			case MSBuildSpecialCommitKind.AttributeValueSpecial:
```

Comment:
How does this differ from how attribute values in general should be committed?

Reply:
This is for the case when you trying to add attributes within the element tags: `<UseWPF Condition=""></UseWPF>` or `<PackageReference Include="Newtonsoft.Json" Version="12.0.2"/>`

It doesn't interfere if you are trying to add attributes like this: [ `<UseWPF>` some attributes `</UseWPF>` ]